cmd/check_spec.go

```diff
@@ -122,58 +92,448 @@ func newCheckSpec(ctx context.Context) (*checkSpec, error) {
+	}
+
+	if err := validateVersion(res.Version.RosettaVersion); err != nil {
+		setValidationStatus(output, version, checkSpecFailure)
+		printError("%v\n", err)
+	}
+
+	if err := validateVersion(res.Version.NodeVersion); err != nil {
+		setValidationStatus(output, version, checkSpecFailure)
+		printError("%v\n", err)
+	}
+
+	// allow is required
+	if res.Allow == nil {
+		setValidationStatus(output, allow, checkSpecFailure)
+		printError("%v: unable to find allow in /network/options response\n", errAllowNullPointer)
+	}
+
+	if err := validateOperationStatuses(res.Allow.OperationStatuses); err != nil {
+		setValidationStatus(output, allow, checkSpecFailure)
+		printError("%v\n", err)
+	}
+
+	if err := validateOperationTypes(res.Allow.OperationTypes); err != nil {
+		setValidationStatus(output, allow, checkSpecFailure)
+		printError("%v\n", err)
+	}
+
+	if err := validateErrors(res.Allow.Errors); err != nil {
+		setValidationStatus(output, allow, checkSpecFailure)
+		printError("%v\n", err)
+	}
+
+	if err := validateCallMethods(res.Allow.CallMethods); err != nil {
+		setValidationStatus(output, allow, checkSpecFailure)
+		printError("%v\n", err)
+	}
+
+	if err := validateBalanceExemptions(res.Allow.BalanceExemptions); err != nil {
+		setValidationStatus(output, allow, checkSpecFailure)
+		printError("%v\n", err)
+	}
+
+	return output
+}
+
+func (cs *checkSpec) networkStatus(ctx context.Context) checkSpecOutput {
+	printInfo("validating /network/status ...\n")
+	output := checkSpecOutput{
+		api: networkStatus,
+		validation: map[checkSpecRequirement]checkSpecStatus{
+			currentBlockID:   checkSpecSuccess,
+			currentBlockTime: checkSpecSuccess,
+			genesisBlockID:   checkSpecSuccess,
+			peers:            checkSpecSuccess,
+		},
+	}
+	defer printInfo("/network/status validated\n")
+
 	res, err := cs.onlineFetcher.NetworkStatusRetry(ctx, Config.Network, nil)
 	if err != nil {
-		return fmt.Errorf("%w: unable to fetch network status", err.Err)
+		printError("%v: unable to fetch network status\n", err.Err)
+		markAllValidationsFailed(output)
+		return output
 	}
 
-	if err := verifyBlockIdentifier(res.CurrentBlockIdentifier); err != nil {
-		return fmt.Errorf("%w", err)
+	// current_block_identifier is required
+	if err := validateBlockIdentifier(res.CurrentBlockIdentifier); err != nil {
+		printError("%v\n", err)
+		setValidationStatus(output, currentBlockID, checkSpecFailure)
 	}
 
-	// TODO
-	return nil
+	// current_block_timestamp is required
+	if err := validateTimestamp(res.CurrentBlockTimestamp); err != nil {
+		printError("%v\n", err)
+		setValidationStatus(output, currentBlockTime, checkSpecFailure)
+	}
+
+	// genesis_block_identifier is required
+	if err := validateBlockIdentifier(res.GenesisBlockIdentifier); err != nil {
+		printError("%v\n", err)
+		setValidationStatus(output, genesisBlockID, checkSpecFailure)
+	}
+
+	// peers is required
+	if err := validatePeers(res.Peers); err != nil {
```

Comment:
So chain does not implement peers.

Reply:
@xiaying-peng sorry it's a statement or a question?

Reply:
this is a statement, I think some chains are unable to get peers

Reply:
Ahh, if that's the case, then I guess we need to update our rosetta doc: https://www.rosetta-api.org/docs/models/NetworkStatusResponse.html, as it says 'peers' is required in /network/status response.